test/e2e/major_upgrade_v2.go

```diff
@@ -0,0 +1,140 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/celestiaorg/celestia-app/v2/app"
+	"github.com/celestiaorg/celestia-app/v2/app/encoding"
+	v1 "github.com/celestiaorg/celestia-app/v2/pkg/appconsts/v1"
+	v2 "github.com/celestiaorg/celestia-app/v2/pkg/appconsts/v2"
+	"github.com/celestiaorg/celestia-app/v2/test/e2e/testnet"
+	"github.com/celestiaorg/celestia-app/v2/test/txsim"
+	"github.com/celestiaorg/knuu/pkg/knuu"
+	"github.com/tendermint/tendermint/rpc/client/http"
+)
+
+func MajorUpgradeToV2(logger *log.Logger) error {
+	latestVersion, err := testnet.GetLatestVersion()
+	testnet.NoError("failed to get latest version", err)
+
+	logger.Println("Running major upgrade to v2 test", "version", latestVersion)
+
+	numNodes := 4
+	upgradeHeight := int64(12)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	logger.Println("Creating testnet")
+	testNet, err := testnet.New("runMajorUpgradeToV2", seed, nil, "test")
+	testnet.NoError("failed to create testnet", err)
+
+	defer testNet.Cleanup()
+
+	testNet.SetConsensusParams(app.DefaultInitialConsensusParams())
+
+	preloader, err := knuu.NewPreloader()
+	testnet.NoError("failed to create preloader", err)
+
+	defer func() { _ = preloader.EmptyImages() }()
+	testnet.NoError("failed to add image", preloader.AddImage(testnet.DockerImageName(latestVersion)))
+
+	logger.Println("Creating genesis nodes")
+	for i := 0; i < numNodes; i++ {
+		err := testNet.CreateGenesisNode(latestVersion, 10000000, upgradeHeight, testnet.DefaultResources)
+		testnet.NoError("failed to create genesis node", err)
+	}
+
+	kr, err := testNet.CreateAccount("alice", 1e12, "")
+	testnet.NoError("failed to create account", err)
```

Comment:
nit for future:

if we move files, ideally (I haven't always done this!) we only move them and not change anything to reduce the effort to see what actually changed. in this case it looked like it was just some comments

Reply:
yeah I tend to agree but since it was only minor changes I applied suggestions